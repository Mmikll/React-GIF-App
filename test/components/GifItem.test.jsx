import { GifItem } from "../../src/components/GifItem"
import { render, screen} from "@testing-library/react"
describe('GifItem Test', () => {
    const title = "titulo"
    const url = "https://localhost/image.jpg"

    test('should be the same as the snapshot', () => {
        
        const {container} = render( <GifItem title={title} url={url} />)

        expect(container).toMatchSnapshot()
    })
    
    test('should return the same URL and Alt', () => {
    
        render( <GifItem title={title} url={url} />)

        const {src, alt} = screen.getByRole('img')

        expect(src).toBe(url)

        expect(alt).toBe(title)

    })
    

    test('should return a text in h1', () => {
        
        render( <GifItem title={title} url={url} />)

        expect(screen.getByText('titulo')).toBeTruthy()

    })
    

})
