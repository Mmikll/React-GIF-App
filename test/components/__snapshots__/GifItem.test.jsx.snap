// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Test should be the same as the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="https://localhost/image.jpg"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
